### DISCUSSION TITLE
Hard refreshing on dynamic routed pages in next.js

### DISCUSSION THREAD

**Discussion by RobinWitt** [neuefische/questions#464] [Web] [answered]
Hello everyone, a lot of times we use dynamic routes in next.js, which means the content of the Page is decided on the current slug/id. 
If you hard refresh (by clicking "refresh" in your browser, for example) said Page, it is possible that an error occurs because the slug/id can't be read at the moment the Page is rerendered.
The solution I found in one of the many StackOverflow answers, was to check `if(router.isReady) ...` before `return`ing the page content. Seems to be working fine.

Any concerns or thoughts on the solution above?
What's your preferred way to handle these "refreshing errors"?

**Comment by NikRoe** [comment, marked as answer]
Hello Robin,

according to the docs from Next.js, you should only work with `router.isReady` within a useEffect. See [this](https://nextjs.org/docs/api-reference/next/router#router-object) as reference.

You are checking this within an if-statement for the moment.

To check whether your content is ready to be displayed, you might check it like this:
```js
if(!slug){
return <div>Loading...</div>
}
```

Hope this helps :) 


On another note: That you have to either check with `router.isReady` or `(!slug)` whether your page is ready to be rendered, might be an indicator that your code can potentially be optimised. Can you send us a link to a PR?

**Reply by RobinWitt** [reply]
Thanks for the quick answer. Good to know that `router.isReady` should not be used in this case. 
I changed it following your suggestion, checking `if(!slug) ...`
If you want to take a look, here is the [Pull Request](https://github.com/RobinWitt/capstone-project/pull/13).
However, I would like to focus on the next user story and think about that (solved) problem later.

**Reply by NikRoe** [reply]
I like your approach to go ahead and work on the next user story!
